From 575e151a548e438c9e97826e3d6add9950648fc6 Mon Sep 17 00:00:00 2001
From: rbuj <robert.buj@gmail.com>
Date: Thu, 18 Apr 2019 23:02:36 +0200
Subject: [PATCH] multiload: filter-out non-local disks and user mounts

Fix #171
---
 multiload/linux-proc.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/multiload/linux-proc.c b/multiload/linux-proc.c
index fcd48bff..c661dd8f 100644
--- a/multiload/linux-proc.c
+++ b/multiload/linux-proc.c
@@ -123,9 +123,10 @@ GetDiskLoad (int Maximum, int data [3], LoadGraph *g)
         struct statvfs statresult;
         glibtop_fsusage fsusage;
 
-        if (strcmp(mountentries[i].type, "smbfs") == 0
-            || strcmp(mountentries[i].type, "nfs") == 0
-            || strcmp(mountentries[i].type, "cifs") == 0)
+        if (strstr (mountentries[i].devname, "/dev/") == NULL)
+            continue;
+
+        if (strstr (mountentries[i].mountdir, "/media/") != NULL)
             continue;
 
         if (statvfs (mountentries[i].mountdir, &statresult) < 0)
-- 
2.21.0

